src/main/java/com/google/graphgeckos/dashboard/storage/DataRepository.java

```diff
@@ -72,7 +75,17 @@
    * @param commitHash the commitHash to search for
    * @return null if no object was found, else a BuildInfo instance of the database entry
    *     associated to that commitHash.
-   * @throws IllegalArgumentException if commitHash is null
    */
-  BuildInfo getRevisionEntry(String commitHash) throws IllegalArgumentException;
+  BuildInfo getRevisionEntry(String commitHash);
+
+  /**
+   * Checks if the given entry is present.
+   *
+   * @param commitHash the commitHash to search for.
+   * @return false if no object was found, else returns true.
+   */
+  default boolean hasRevisionEntry(String commitHash) {
```

Comment:
I do not think this has to be specified in the interface, since it is only used by the DatastoreRepository itself.

Reply:
It's quit nice to have default functions in interfaces if we know, that all inherited classes have exactly same implementation of this function.

Reply:
And how do we know they have the exact same implementation? For the scope of this project, we only need one implementation of this.

Reply:
Interfaces are used to make code more extensible among other possibilities they provide, this function will obviously be exactly same for all inherited classes because it uses getRevisionEntry to check if the element it returns is null. If the implementations changes this function could be simple overwritten.